lib/handler/fastcgi.c

```diff
@@ -628,6 +628,11 @@ static int handle_stdin_record(struct st_fcgi_generator_t *generator, struct st_
         h2o_buffer_consume(&generator->resp.receiving, parse_result);
     }
 
+    if (generator->resp.receiving->size == 0) {
+        h2o_doublebuffer_prepare_empty(&generator->resp.sending);
+        h2o_send(generator->req, NULL, 0, H2O_SEND_STATE_IN_PROGRESS);
```

Comment:
Would it be better if we placed this code at the end of `on_read`?

I would anticipate that it would be natural for a FastCGI application to encode the headers in first FCGI_STDOUT record and body in the next record, and send at once.

Reply:
https://github.com/h2o/h2o/pull/1508/commits/a393f90f20b62160667dc6ad972d4bb8d40640c1